fixes contrast in color correction and filters, cleans up datepicker

designmatty opened the pull request:
## Overview
There were contrast issues on all three themes pertaining to the advanced color correction panel and scene filter panel. Beyond fixing these contrast issues, I spent some time to clean up other interactions.

- Cleaned up the UI for the date picker
- In advanced color correction, each label appearing next checkbox will now activate the checkbox when clicked.
- Each "select box" for the various dropdowns in the filters panel should look roughly the same now.
- The labels for each section in the filter panel should now be sitting above its respective filter (rather than to the left).

### Checklist

- [ ] Description of PR is in an appropriate section of the [changelog](https://github.com/raster-foundry/raster-foundry/blob/develop/CHANGELOG.md) and grouped with similar changes if possible
~- [ ] Styleguide updated, if necessary~
~- [ ] [Swagger specification](https://github.com/raster-foundry/raster-foundry-api-spec) updated~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~
~- [ ] Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~
~- [ ] Any new SQL strings have tests~

### Demo

<img width="396" alt="screen shot 2018-11-05 at 11 22 40 am" src="https://user-images.githubusercontent.com/1928955/48012756-bd72f280-e0f0-11e8-8679-7dc3ba5b7428.png">
<img width="400" alt="screen shot 2018-11-05 at 11 23 16 am" src="https://user-images.githubusercontent.com/1928955/48012757-bd72f280-e0f0-11e8-8282-16704f81aa7f.png">
<img width="414" alt="screen shot 2018-11-05 at 11 52 32 am" src="https://user-images.githubusercontent.com/1928955/48013013-573a9f80-e0f1-11e8-922d-c034a999ca0a.png">
<img width="397" alt="screen shot 2018-11-05 at 11 25 44 am" src="https://user-images.githubusercontent.com/1928955/48012759-bd72f280-e0f0-11e8-9f25-640c22c8faef.png">
<img width="402" alt="screen shot 2018-11-05 at 11 27 40 am" src="https://user-images.githubusercontent.com/1928955/48012760-bd72f280-e0f0-11e8-8949-99180048dda8.png">
<img width="397" alt="screen shot 2018-11-05 at 11 35 20 am" src="https://user-images.githubusercontent.com/1928955/48012761-bd72f280-e0f0-11e8-89f2-d8011e97f53a.png">

## Testing Instructions

 * Loop through each theme to make sure the following lines up with the above screens:
   * Project browse filters panel
   * Project advanced color correction pane

Closes #4243

alkamin commented on app-frontend/src/assets/styles/sass/components/_searchselect.scss:
```diff
@@ -32,21 +32,23 @@
-  color: $gray;
-  border: 1px solid $gray-lighter;
-  background: $gray-lighter;
+  color: $white;
```
Not sure where exactly this is happening css wise but the datasource select menu is covering the input which make using the filtering option a bit difficult.

![image](https://user-images.githubusercontent.com/2442245/48154798-81c65d00-e297-11e8-8e8c-7a27a121ebb0.png)

designmatty commented:
- **Updates the dropdowns within the filter pane to address @alkamin's concerns:**
<img width="383" alt="screen shot 2018-11-16 at 2 37 26 pm" src="https://user-images.githubusercontent.com/1928955/48646467-edef4200-e9b6-11e8-8fff-cac58f8318ac.png">
<img width="386" alt="screen shot 2018-11-16 at 3 46 22 pm" src="https://user-images.githubusercontent.com/1928955/48646468-edef4200-e9b6-11e8-9037-ef02b502c5ca.png">
<img width="388" alt="screen shot 2018-11-16 at 3 46 27 pm" src="https://user-images.githubusercontent.com/1928955/48646469-edef4200-e9b6-11e8-952c-3a212f4d8718.png">
<img width="389" alt="screen shot 2018-11-16 at 3 46 34 pm" src="https://user-images.githubusercontent.com/1928955/48646470-ee87d880-e9b6-11e8-95f8-66f25395a055.png">

- Changed default min-width of all dropdown's
  - required a slight adjustment to how we rendered sidebar header controls
- Fixed a regression that hid the clear button from certain filter's
- Removed odd styling decisions I made years ago on scene meta data editing modal

designmatty commented:
@alkamin this is ready for another review
